### ISSUE TITLE
why the same embedding result when two input image?

### ISSUE THREAD

**Issue by guoxiaolu** [VisualComputingInstitute/triplet-reid#34]
`img = cv2.imread('./test/004.jpg')
img = cv2.resize(img, (128,256))
x = np.expand_dims(img,axis=0)

sess = tf.Session()
saver = tf.train.import_meta_graph('market1501_weights/checkpoint-25000.meta')
saver.restore(sess, 'market1501_weights/checkpoint-25000')

input_x = sess.graph.get_tensor_by_name("sub:0")
emb_weights = sess.graph.get_tensor_by_name('emb/weights:0')
feature1 = sess.run(emb_weights, feed_dict={input_x: x})
print feature1`

I have tested the tensorflow code using pretrained model, but it outputs the same result when I tested it on two images. I don't know why. Are there some errors in my code?

Besides, I have tested the pytorch version, and the embedding shape is (1,2048) but your code is (1024, 128). To my knowledge, the traditional embedding features shape is like pytorch version. Is there some difference in both version?

**Comment by lucasb-eyer** [comment]
That's because you are getting the weights of the the last layer, not the actual embeddings. I'm not sure if it's possible to load the model like you're doing. The canonical way is to do like we do in `embed.py`.

**Comment by guoxiaolu** [comment]
okay, thank you for your reply. I have checked the embed.py and it works correctly when inference using the given canonical way. Thank you very much, good job~

**Comment by maxisme** [comment]
@guoxiaolu have you got it working? I am also wanting to start the session like this for deployment reasons.

**Comment by Pandoro** [comment]
Just to comment and point this out to future readers: `embed.py` IS our inference code. Everything you did was just copy some of the arguments that are "hidden" in `args.json` into the code. We specifically designed things to be "hidden" and also check these, since certain setups do not make any sense. For example if I train a resnet_v1_50, it doesn't make sense to provide a different model during inference.

**Comment by guoxiaolu** [comment]
Okay, Thank you very much~
